packit/copr_helper.py

```diff
@@ -134,6 +110,95 @@ def create_copr_project_if_not_exists(
                 additional_packages=additional_packages,
                 additional_repos=additional_repos,
             )
+            return
+
+        delete_after_days = -1 if preserve_project else 60
+
+        fields_to_change = self.get_fields_to_change(
+            copr_proj=copr_proj,
+            additional_repos=additional_repos,
+            chroots=chroots,
+            description=description,
+            instructions=instructions,
+            list_on_homepage=list_on_homepage,
+            delete_after_days=delete_after_days,
+        )
+
+        if fields_to_change:
```

Comment:
Aaaaand, one more thing :)

If I understand it correctly, the aim of this change is to edit the Copr project only of one of the setting fields have changed. Is this right?

How will this work in the context of how the convert2rhel team is using Packit?

They have their own repo set up, with an already existing `description`, `delete_after_days`, `instructions` and `list_on_homepage` setting, but they only specify `owner`, `project` and the `chroot`s they want to build for in their `packit.yaml`.

Will packit try to change the settings of their Copr repo, given the above? Is that the thing we want?

Reply:
> If I understand it correctly, the aim of this change is to edit the Copr project only of one of the setting fields have changed. Is this right?

:+1:

> How will this work in the context of how the convert2rhel team is using Packit?
> Will packit try to change the settings of their Copr repo, given the above? Is that the thing we want?

They changed chroots already => they will receive a request for admin permissions. And
- If they change the settings manually, we will be fine next time.
- If they approve the request, we will update the settings by ourselves next time.

And also next time if we find some difference between the config file and the project's settings.

Reply:
I see.

Will this mean, that users will have to keep their Copr project settings in sync with `packit.yaml` if they don't want to be bothered by these notifications?

Does it make sense to ask users using their own Copr repos to also specify repository description and all kinds of other data in their `packit.yaml`? Does this configuration living in `packit.yaml` strictly relate to what Packit's main functionality is? I feel that handling these settings is way out of the skope of what Packit is expected to do.

Packit should be able to trigger builds in Copr and needs some permissions for that, but managing Copr projects settings seems out of scope to me.

My main concern here is: as soon as we start requesting admin permissions for Copr projects owned by others, we take on the risk of ruining those projects and making our users very angry. We can discuss how big of a risk this is, but if there is an option to avoid it completely, I would prefer going that way.

(I'll need to read up what that additional_repos request was about...)

Reply:
> Will this mean, that users will have to keep their Copr project settings in sync with `packit.yaml` if they don't want to be bothered by these notifications?

No, if:
- they give us admin rights
- or don't change the settings in the config

> Does it make sense to ask users using their own Copr repos to also specify repository description and all kinds of other data in their packit.yaml?

We can do it, it is prepared for that case but we don't have support in the config.

> Does this configuration living in packit.yaml strictly relate to what Packit's main functionality is? I feel that handling these settings is way out of the skope of what Packit is expected to do.

Depends on the specific value. Description? Maybe not. Additional repos? Definitely. Someone can't build without that.
We are providing automation for integration between upstream and Fedora -- creating preservable copr repos with configurable description fits well to this.

> My main concern here is: as soon as we start requesting admin permissions for Copr projects owned by others, we take on the risk of ruining those projects and making our users very angry. We can discuss how big of a risk this is, but if there is an option to avoid it completely, I would prefer going that way.
> Packit should be able to trigger builds in Copr and needs some permissions for that, but managing Copr projects settings seems out of scope to me.

If we want to support custom projects, we still need to update chroots now and then. And they have a choice -- give us the rights, do it manually or use our user+project and don't care at all.

I feel that risk but can't find a better way..